shell/gpu/gpu_surface_metal_skia.mm

```diff
@@ -48,7 +48,7 @@
                                                SkSurfaces::TextureReleaseProc release_proc,
                                                SkSurface::ReleaseContext release_context) {
   GrMtlTextureInfo info;
-  info.fTexture.reset([texture retain]);
+  info.fTexture.reset((__bridge_retained GrMTLHandle)texture);
```

Comment:
What do you think about changing this to `retain` and `__bridge`? I feel like it's easier in an ARC world to make the smart pointer explicitly manage all of the retains whenever possible, rather than using the more complex `__bridge_*` to manually change ARC's retains.

Reply:
I had originally done that (because it reads better at the call site) but the semantics are very slightly different:
* `sk_cfp::retain` takes a non-ARC-managed pointer and calls `CFRetain` on it.
* `sk_cfp::reset` takes a non-ARC-managed pointer and does not call `CFRetain` on it (you have to do it yourself).

Neither of these C++ methods is annotated with ARC attributes. I actually see segfaults if I switch to `retain` and `__bridge`.

Reply:
> but the semantics are very slightly different:
> 
> * `sk_cfp::retain` takes a non-ARC-managed pointer and calls `CFRetain` on it.
> * `sk_cfp::reset` takes a non-ARC-managed pointer and does not call `CFRetain` on it (you have to do it yourself).

Right, but switching from `__bridge_retained` to `__bridge` offsets that the other way. What's the issue? I'm not clear on why the `__bridge`+`retain()` version would crash if this version doesn't.

> Neither of these C++ methods is annotated with ARC attributes.

Right, but since we are passing in bridged pointers that shouldn't matter, right? Either way, they are already out of ARC's hands before they become method arguments. That's why we need the bridge calls in the first place; if they took ARC-manageable pointer types they would interoperate.

Reply:
Agreed -- in my original patch I'd actually migrated all the `reset(__bridge_retained)` to `retain(__bridge)` but ended up hitting a crash and had made an assumption that somehow the compiler had found a way to aggressively nil the pointer.

However, on further digging today this was due to an error on my part elsewhere in the code, making a strong pointer weak that I shouldn't have. That's now fixed and I've migrated all `reset(__bridge_retained)` to `retain(__bridge)`.